x/mongo/driver/operation.go

```diff
@@ -697,6 +708,10 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 				// If we got a retryable error or MaxTimeMSExpired error, we add UnknownTransactionCommitResult.
 				tt.Labels = append(tt.Labels, UnknownTransactionCommitResult)
 			}
+
+			if tt.HasErrorLabel(NoWritesPerformed) {
```

Comment:
I'm surprised this is under the `Error` `case`, as it seems like you're only _setting_ `prevIndefiniteError` for `WriteCommandError`s on L382. Again, I'm wondering if `NoWritesPerformed` can appear on both `driver.Error`s and `driver.WriteCommandError`s. If so, I think we need this check in both of the cases for this switch.

Reply:
We are only setting the prevIndefiniteError for a write command. The work flow should look something like this:

1. WriteCommandError passed from server
2. Enter WriteCommandSwitch
3. Enter resetForRetry 
4. Set previousIndefiniteError
5. Retry happens
6. Error with NoWritePerfomed occurs
7. Return previousIndefiniteError